website/versioned_docs/version-22.4/WatchPlugins.md

```diff
@@ -0,0 +1,125 @@
+---
+id: version-22.4-watch-plugins
+title: Watch Plugins
+original_id: watch-plugins
+---
+
+A watch plugin allows you to enhance Jest in a couple of ways. It provides you with ways of hooking into specific parts of Jest. It allows you to handle user input and provide an interactive experience
+
+## JestHooks
+
+Jest hooks can be accessed by implementing the `apply` method in a plugin
+class. This receives a `jestHooks` argument that allows the plugin to hook into
+specific parts of the lifecycle of a test run.
+
+```javascript
```

Comment:
What do you think about moving this up to it's own section "Plugin Class" that includes the whole interface?